### PR TITLE
Feat/NC Medicaid Sub Categories

### PR DESCRIPTION
**Pull request by msrezaie**
What (if any) features are you implementing?
- Added Medicaid sub category calculations for North Carolina. Eligibility rules taken from https://policies.ncdhhs.gov/wp-content/uploads/MEDICAID-ELIGIBILITY_9-9-2024.pdf
- Added a sperate program for NC Emergency Medicaid. This program's Legal Status needs to only include "non-citizen" (through the admin dashboard) in order for it to only show up with the immigration filter set to "Individuals without lawful U.S. presence or citizenship" (as per the program's eligibility [rules](https://policies.ncdhhs.gov/wp-content/uploads/ma-3240-5.pdf))

Edit:
What (if anything) did you refactor?
- The original approach for determining a household member's eligibility for a Medicaid sub category was done manually. But with the use of `medicaid_category` and `is_optional_senior_or_disabled_for_medicaid` Policy Engine variables, a household member's Medicaid eligibility is now done through PE.

### CODE REVIEW

**Comment by msrezaie** [comment]
Linking the test document with different household scenarios:
https://docs.google.com/document/d/14uT3qhqc_3I7HvBmEgyLQguMfWT4-u0ssUngs3EbATA/edit?tab=t.0

File: programs/programs/nc/medicaid/emergency_medicaid/calculator.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    def household_eligible(self, e: Eligibility):
+        # Does not have insurance
+        has_no_insurance = False
+        for member in self.screen.household_members.all():
+            has_no_insurance = member.insurance.has_insurance_types(("none",)) or has_no_insurance
+
+            # Pregnant and under 18 years old have a different FPL percentage
+            if member.age <= 18 and member.pregnant:
+                EmergencyMedicaid.fpl_percent = 2.11
```
Comment (CalebPena) [inline programs/programs/nc/medicaid/emergency_medicaid/calculator.py:21]: This is permanently changing the `fpl_percent` on this class.

File: programs/programs/nc/medicaid/emergency_medicaid/calculator.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+from programs.programs.helpers import medicaid_eligible
+import programs.programs.messages as messages
+
+
+class EmergencyMedicaid(ProgramCalculator):
+    # $6,268/yr | ~$522/mo
+    member_amount = 6268
+    max_age = 64
+    fpl_percent = 1.96
+    dependencies = ["age", "insurance", "income_amount", "household_size"]
```
Comment (CalebPena) [inline programs/programs/nc/medicaid/emergency_medicaid/calculator.py:11]:
```suggestion
    dependencies = ["age", "insurance", "income_amount", "income_frequency", "household_size"]
```

**Review by CalebPena — APPROVED** [review]
Looks good after a couple of easy fixes.

File: programs/programs/nc/medicaid/emergency_medicaid/calculator.py (hunk 1 above)
Comment (msrezaie) [reply programs/programs/nc/medicaid/emergency_medicaid/calculator.py]: good catch, thanks

**Pull request merged**